src/nodes.coffee

```diff
@@ -2140,6 +2154,9 @@ exports.Code = class Code extends Base
     wasEmpty = @body.isEmpty()
     @body.expressions.unshift thisAssignments... unless @expandCtorSuper thisAssignments
     @body.expressions.unshift exprs...
+    if @isMethod and @bound and not @isStatic and @constructorName
```

Comment:
The addition of the actual runtime check (using new `_boundMethodCheck` helper). Currently deals with anonymous classes via option (2) (forgo the runtime check for anonymous classes) with `and @constructorName`